warp.m

```diff
@@ -88,12 +92,37 @@
 for i = 12:1:m
     for j = max([round(i/3) (i-m)*3+n]):min([3*i round((i-m)/3+n)])
         if k(i-1,j) >= 3
-            [y x] = max([N(i-1,j-1) NaN N(i-1, j-2) N(i-1, j-3)]);
+            y = max(max(N(i-1,j-1), NaN), max(N(i-1, j-2), N(i-1, j-3)));
+            ks = 1;
+            if y == N(i-1,j-1)
+                x = 1;
+            elseif isnan(y)
+                x = 2;
+                ks = 1 + k(i-1,j);
+            elseif y == N(i-1, j - 2)
+                x = 3;
+            elseif y == N(i-1, j -3)
+                x = 4;
+            end
+            % [y x] = max([N(i-1,j-1) NaN N(i-1, j-2) N(i-1, j-3)]);
         else
-            [y x] = max([N(i-1,j-1) N(i-1,j) N(i-1, j-2) N(i-1, j-3)]);
+            y = max(max(N(i-1,j-1), N(i-1,j)), max(N(i-1, j-2), N(i-1, j-3)));
+            ks = 1;
+            if y == N(i-1,j-1)
+                x = 1;
+            elseif y == N(i-1,j)
+                x = 2;
+                ks = 1 + k(i-1,j);
+            elseif y == N(i-1, j - 2)
+                x = 3;
+            elseif y == N(i-1, j -3)
+                x = 4;
+            end
+            %  [y x] = max([N(i-1,j-1) N(i-1,j) N(i-1, j-2) N(i-1, j-3)]);
         end
-        ks = [1 1+k(i-1,j) 1 1];
-        k(i,j) = ks(x);
+    %    ks = [1 1+k(i-1,j) 1 1];
```

Comment:
@RohanK22 what is the purpose of these commented out lines? Are they comments explaining what the code is doing? Or old code commented out? Would be good to make it clear in the code, it will be hard to remember after a while.

Reply:
@alex-konovalov, yes those comments were old code that I kept for reference while writing new lines. I've modified the commit and removed the unwanted comments. Let me know of any further changes required.